adsmp/app.py

```diff
@@ -160,3 +217,59 @@ def get_msg_status(self, msg):
                 return 'active'
         else:
             return 'unknown'
+        
+    
+
+    def reindex(self, solr_docs, batch_insert, batch_update, solr_urls):
+        """Sends documents to solr and to Metrics DB.
+        
+        :param: solr_docs - list of json objects (solr documents)
+        :param: batch_insert - list of json objects to insert into the metrics
+        :param: batch_update - list of json objects to update in metrics db
+        :param: solr_urls - list of strings, solr servers.
+        """
+        
+        out = solr_updater.update_solr(solr_docs, solr_urls, ignore_errors=True)
+        errs = [x for x in out if x != 200]
+        
+        if len(errs) == 0:
+            self._mark_processed(solr_docs)
+            self._insert_into_metrics_db(batch_insert, batch_update)
+        else:
+            # recover from erros by inserting docs one by one
+            for doc in solr_docs:
```

Comment:
we can't, the batch was sent as one large stream; if it failed on 10th doc, we can't assume that 11th or 99th will succeed